### PR TITLE
refactor(client-services): signal manager status

### PR DESCRIPTION
**Pull request by wittjosiah**
### Summary
🔄📡🕸️

The pull request refactors and enhances the network and signal management logic for the client services and the devtools. It uses observables, streams, and enums from `@dxos/async` and `@dxos/protocols` to simplify the state management and avoid duplication and race conditions. It also introduces interfaces and factories for the signal managers and the transports, which allow testing and configuring different implementations of these components. It updates the network service API and the devtools UI to use the new terminology and formats for the network and signal status and configuration. It modifies several files in `packages/core/mesh`, `packages/core/protocols`, `packages/sdk/client-services`, `packages/devtools/devtools`, and `packages/experimental/kai`.

> _To make the network service more robust_
> _We refactored the code with some gust_
> _We used `SignalManager` and `TransportFactory`_
> _And observables from `@dxos/async`_
> _And enums from `@dxos/protocols` we trust_

### Walkthrough
*  Replace the `statusChanged` event and the `_state` field of the `SignalClient` class with a `state` observable and a `_pushState` stream, using the `SignalState` enum from `@dxos/protocols` ([link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL7-R17),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL13-R23),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL24-L43),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL68-R58),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL97-R87),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL158-R148),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL164-R157),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL184-R174),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL198-R196),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL218-R212),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL245-R239),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL273-R272),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL309-R302),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL316-R308),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL324-R318),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL336-R328),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL351-R351),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL371-R363),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL380-R372),[link](https://github.com/dxos/dxos/pull/3120/files?diff=unified&w=0#diff-ce6f3b0b5070c5323d58e3f92d0e262f3a2f39e82d550ffc4bd47c4857f44a2fL420-R412)).

### CODE REVIEW

**Comment by wittjosiah** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3120** <a href="https://app.graphite.dev/github/pr/dxos/dxos/3120" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/3120?utm_source=stack-comment).

File: packages/core/mesh/messaging/src/signal-client/signal-client.ts [hunk 1]
```diff
@@ -65,7 +55,7 @@ export type CommandTrace = {
  * Establishes a websocket connection to signal server and provides RPC methods.
  */
 export class SignalClient implements SignalMethods {
-  private _state = SignalState.CLOSED;
+  private readonly _pushState = new PushStream<SignalState>();
```
Comment (dmaretskyi) [inline packages/core/mesh/messaging/src/signal-client/signal-client.ts:58]: not sure about observables in core, on one hand coupling events with state is nice, but on the other hand the flexibility of having separate events feels more important.

Comment (dmaretskyi) [reply packages/core/mesh/messaging/src/signal-client/signal-client.ts:58]: although this example does look better with observables (its just one value coupled with it's update event), selectively using observables in core packages might add confusion on whether a specific case should be handled by events or observables. in this case for example, we might want to replace `stateChanged` with `connected` `closed`, and `error` events, which would be hard with observables

Comment (dmaretskyi) [reply packages/core/mesh/messaging/src/signal-client/signal-client.ts:58]: overall, obserables are a great fit in the client sdk, where the need to make api consistent and bulletproof is higher. In lower-level code I think flexibility and granularity is more important, and I'd prefer to stick to events

Comment (wittjosiah) [reply packages/core/mesh/messaging/src/signal-client/signal-client.ts:58]: Yesterday I thought it was going to be necessary or at least make it far easier to hook up for some reason, but that didn't turn out to be the case. I think the implementations are almost identical in this case, so yeah, I'll just switch it back.

**Pull request merged**